### PR TITLE
backport pr 15979 into 1.14.x

### PR DESCRIPTION
**Pull request by malizz**
manually backport pr #15979 into 1.14.x (backport assist failed)
This PR update protobuf package to `google.golang.org/protobuf` since `github.com/golang/protobuf` is deprecated. 

### PR Checklist

* [x] updated test coverage
* [NA] external facing docs updated
* [ ] appropriate backport labels added
* [x] not a security concern

### CODE REVIEW

File: .changelog/15979.txt
Comment (DanStough) [inline .changelog/15979.txt:1]: Not blocking, but you may want to change this to #17287 if you are changing other things.

File: agent/xds/config.go [hunk 1]
```diff
@@ -192,8 +193,15 @@ func ToOutlierDetection(p *structs.PassiveHealthCheck) *envoy_cluster_v3.Outlier
 	}
 
 	if p.EnforcingConsecutive5xx != nil {
+		// NOTE: EnforcingConsecutive5xx must be greater than 0 for ingress-gateway
 		od.EnforcingConsecutive_5Xx = &wrappers.UInt32Value{Value: *p.EnforcingConsecutive5xx}
```
Comment (DanStough) [inline agent/xds/config.go:197]: It looks like `wrappers` comes from the deprecated GH package. I don't think it's critical to this PR, but if we went through the process of upgrading the proto generation to the `google.golang` version, we might want to clean it out completely. Not sure how much of a lift this is, though. Not blocking.

**Review by DanStough — APPROVED** [review]
A couple non-blocking comments. Thank you for this cleanup!

File: agent/xds/config.go (hunk 1 above)
Comment (malizz) [reply agent/xds/config.go:197]: Good point, I can change it in another PR. I want to make sure it won't break the backport assist PRs with out of band changes.

**Pull request merged**